tox.ini

```diff
@@ -317,40 +309,25 @@ commands =
     ; bash -c '(docker logs test_domain_2-backend-1 -f &) | grep -q "Application startup complete" || true'
     bash -c '(docker logs test-gateway-1-backend-1 -f &) | grep -q "Application startup complete" || true'
 
-    ; frontend
-    bash -c 'if [[ "$PYTEST_MODULES" == *"frontend"* ]]; then \
-        echo "Starting frontend"; date; \
-        pytest tests/integration -m frontend -p no:randomly --co; \
-        pytest tests/integration -m frontend -vvvv -p no:randomly -p no:benchmark -o log_cli=True --capture=no; \
-        return=$?; \
-        docker stop test-domain-1-frontend-1 || true; \
-        echo "Finished frontend"; date; \
-        exit $return; \
-    fi'
-
-    ; network
-    bash -c 'if [[ "$PYTEST_MODULES" == *"network"* ]]; then \
-        echo "Starting network"; date; \
-        pytest tests/integration -m network -p no:randomly --co; \
-        pytest tests/integration -m network -vvvv -p no:randomly -p no:benchmark -o log_cli=True --capture=no; \
-        return=$?; \
-        echo "Finished network"; date; \
-        exit $return; \
-    fi'
-
-    ; container workload
-    bash -c 'if [[ "$PYTEST_MODULES" == *"container_workload"* ]]; then \
-        echo "Starting Container Workload test"; date; \
-        pytest tests/integration -m container_workload -p no:randomly --co; \
-        pytest tests/integration -m container_workload -vvvv -p no:randomly -p no:benchmark -o log_cli=True --capture=no; \
-        return=$?; \
-        echo "Finished container workload"; date; \
-        exit $return; \
-    fi'
+    bash -c '\
+        PYTEST_MODULES=($PYTEST_MODULES); \
+        for i in "${PYTEST_MODULES[@]}"; do \
+            echo "Starting test for $i"; date; \
```

Comment:
stupid question, but which component is responsible now for starting the stack?

Reply:
this will just run the pytest in a more concise way. The stack starts using hagrid - https://github.com/OpenMined/PySyft/pull/8553/files/667c377c2fcbc7604b6b326182383c095fa480a3#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449R300-R302